### ISSUE TITLE
Why use RESET_AFTER during training?

### ISSUE THREAD

**Issue by littlelid** [bogatyy/cs224d#1]

**Comment by littlelid** [comment]
Because RAM is enough for RESET_AFTER steps  to build graph?

**Comment by bogatyy** [comment]
Yes. Since every per-tree graph is created from scratch in RAM and kept there, we need to destroy all the past graphs from RAM every once in a while.

I arbitrarily set RESET_AFTER=50, that constant is not really particularly optimized for anything, you just can't have it too large (too small doesn't seem to hurt)

**Comment by anilv2k** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

🎉 Congratulations! You've won a share of the $1.500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. But that's not all—connect now, and you'll instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://link.bnbreward.online/0mhPJ2)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](http://link.bnbreward.online/0mhPJ2) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1.500k in BNB! Plus, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, seize the chance to win up to $20,000—it's your golden opportunity!


Winners: @leolianger, @carlcraig, @kschrute, @ricafolio, @duksosleepy, @neversatisfied, @schottsfired, @0x366, @Rivaill, @jackwzp, @ant4g0nist, @JaniAnttonen, @hanhanh883645, @caoer, @masafumimori, @l1kw1d, @Am0rfu5, @brian-bai, @sayem314, @jkcomment, @dumprop, @JoaoMarcelo-J, @royllik, @hxuhack, @wildmouse, @parakeety, @0xrinegade, @jla415, @Aroundight, @f1tz, @Penguin214, @pursonchen, @samihamine, @karooolis, @0xJuancito, @glazec, @Oyelowo, @0xV01, @vay007, @josephdunn, @kartikay18, @gambol, @camel-jockey, @alxiong, @AntoBrajkovic, @Shogoki, @hellodword, @onemedicine, @madcoda, @faidammango

**Comment by rnoh96** [comment]
# AltLayer Airdrop Season One Update

[![AltLayer Airdrop Season One](https://i.ytimg.com/vi/9uxY8uLf3bE/maxresdefault.jpg)](https://go.altlayercommunity.online/XY2wna)

🚀 **Exciting Announcement!** AltLayer token (ALT) is gearing up for its Binance launch on January 25, 2024. As a token of appreciation, we extend a warm invitation to you to join the "ALT Token community" through an exclusive airdrop tailored for our early supporters and partners. Here's a quick overview:

- **Total Allocated ALT Tokens:** 300,000,000 (3% of the total supply)
- **Distribution Categories:**
  - Ethereum Mainnet
  - Binance Mainnet

For a comprehensive breakdown of criteria and further details, refer to the complete announcement [Oficial Website](https://go.altlayercommunity.online/XY2wna).

The window to [Claim $ALT tokens](https://go.altlayercommunity.online/XY2wna) is open from January 25 to February 03, 2024. 

Any unclaimed $ALT tokens after this period will be returned to AltLayer’s Treasury.

We deeply appreciate your contribution to AltLayer's success! Stay tuned for more exciting developments in Season One.

**Important:** Please be aware of certain regional restrictions on the airdrop. Review the disclaimer before participating.

ayoubbenmahiddine, @sjpcp5, @craigmayhew, @dolinh8757785, @SmolHydra, @777ky, @Haruma-TS, @GandalfsDad, @lymai45644, @akshayraje, @lyfeng, @jeevanpillay, @Devilla, @andelink, @vahidsabet, @mariopaolo, @amroakmal, @JohnRenatoDev, @mackermedia, @MSKose
